packages/editor/src/components/post-status/index.js

```diff
@@ -0,0 +1,318 @@
+			statusLabel = __( 'Draft' );
+			break;
+		case 'pending':
+			statusLabel = __( 'Pending review' );
+			break;
+		case 'private':
+			statusLabel = __( 'Published privately' );
+			break;
+	}
+	return (
+		<Text
+			className={ classnames( 'editor-post-status-label', {
+				[ ` has-status-${ status }` ]: !! status,
+			} ) }
+		>
+			{ statusLabel }
+		</Text>
+	);
+}
+
+const STATUS_OPTIONS = [
+	{
+		label: (
+			<>
+				{ __( 'Draft' ) }
+				<Text variant="muted" size={ 12 }>
+					{ __( 'Not ready to publish.' ) }
+				</Text>
+			</>
+		),
+		value: 'draft',
+	},
+	{
+		label: (
+			<>
+				{ __( 'Pending' ) }
+				<Text variant="muted" size={ 12 }>
+					{ __( 'Waiting for review before publishing.' ) }
+				</Text>
+			</>
+		),
+		value: 'pending',
+	},
+	{
+		label: (
+			<>
+				{ __( 'Private' ) }
+				<Text variant="muted" size={ 12 }>
+					{ __( 'Only visible to site admins and editors.' ) }
+				</Text>
+			</>
+		),
+		value: 'private',
+	},
+	{
+		label: (
+			<>
+				{ __( 'Scheduled' ) }
+				<Text variant="muted" size={ 12 }>
+					{ __( 'Publish automatically on a chosen date.' ) }
+				</Text>
+			</>
+		),
+		value: 'future',
+	},
+	{
+		label: (
+			<>
+				{ __( 'Published' ) }
+				<Text variant="muted" size={ 12 }>
+					{ __( 'Visible to everyone.' ) }
+				</Text>
+			</>
+		),
+		value: 'publish',
+	},
+];
+
+export default function PostStatus() {
+	const {
+		status,
+		date,
+		password,
+		postId,
+		postType,
+		isBeingScheduled,
+		canEdit,
+	} = useSelect( ( select ) => {
+		const {
+			getEditedPostAttribute,
+			getCurrentPostId,
+			getCurrentPostType,
+			isEditedPostBeingScheduled,
+			getCurrentPost,
+		} = select( editorStore );
+		// const post = select( editorStore ).getCurrentPost();
+		return {
+			status: getEditedPostAttribute( 'status' ),
+			date: getEditedPostAttribute( 'date' ),
+			password: getEditedPostAttribute( 'password' ),
+			postId: getCurrentPostId(),
+			postType: getCurrentPostType(),
+			isBeingScheduled: isEditedPostBeingScheduled(),
+			canEdit: getCurrentPost()._links?.[ 'wp:action-publish' ] ?? false,
+		};
+	}, [] );
+	const [ showPassword, setShowPassword ] = useState( !! password );
+	const passwordInputId = useInstanceId(
+		PostStatus,
+		'editor-change-status__password-input'
+	);
+	const { editEntityRecord } = useDispatch( coreStore );
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const [ popoverAnchor, setPopoverAnchor ] = useState( null );
+	// Memoize popoverProps to avoid returning a new object every time.
+	const popoverProps = useMemo(
+		() => ( {
+			// Anchor the popover to the middle of the entire row so that it doesn't
+			// move around when the label changes.
+			anchor: popoverAnchor,
+			'aria-label': __( 'Status & visibility' ),
+			headerTitle: __( 'Status & visibility' ),
+			placement: 'left-start',
+			offset: 36,
+			shift: true,
+		} ),
+		[ popoverAnchor ]
+	);
+
+	// Effect to update post status when post date changes.
```

Comment:
I'm not sure that we should be updating the status when the date changes. I think we should just update what we show to the user but avoid and "edit" call (create undo level, requires removing the edit when you actually change the date again...). This is probably what's causing the bugs that I was seeing.